lib/secure_tests.js

```diff
@@ -29,6 +29,8 @@ describe('Secure API', function() {
   });
 
   it('should generate private and public RSA key with keysize 2048', function(done) {
+    // Very demanding operation, locally takes cca 5 seconds
+    this.timeout(10000);
```

Comment:
After running the tests multiple times locally, this timeout doesn't seem to be enough. I got a full passing suite 1/5 times.
Also, the 1024 size key test times out too intermittently (after 2 seconds).

Maybe we can update the timeouts to 5 seconds for the 1024 key size (just in case) and 20 seconds for the 2048 key size?
I think its more important that these tests act as regression rather than having a performance element to them